### DISCUSSION TITLE
Zigbee sensor reporting intervals

### DISCUSSION THREAD

**Discussion by summongod** [arendst/Tasmota#10418] [Q&A] [answered]
Will Tasmota support changing the interval period or sensitivity for sensors such as the Sonoff SNZB-02?
At the moment I get a lot of activity due to the sensitivity of the humidity readings and I'm concerned that it will drain the battery.
I have seen this thread https://github.com/Koenkk/zigbee2mqtt/issues/3794 in Zigbee2mqtt and before the change was accepted I made the changes suggested to reduce humidity reporting on my Sonoffs when using ZbBridge with Zigbee2mqtt and it appeared to work as expected.

**Comment by s-hadinger** [comment]
What is the frequency of reporting? Can you share logs of consecutive readings?

**Reply by summongod** [reply]
[log.txt](https://github.com/arendst/Tasmota/files/5771287/log.txt)

**Comment by s-hadinger** [comment, marked as answer]
Thanks. It seems to follow the Zigbee standard, which means you can ask the device to report between `min` and `max` interval in seconds or whenever a change exceeds `x` in value.

As you can see in the logs, you have an example of syntax:
```
20:01:41.098 ZIG: auto-bind `ZbSend {"Device":"0x540C","Config":{"Temperature":{"MinInterval":30,"MaxInterval":3600,"ReportableChange":0.2}}}`
```

In the example above, `ZbSend {"Device":"0x540C","Config":{"Temperature":{"MinInterval":30,"MaxInterval":3600,"ReportableChange":0.2}}}`sets the following:
- don't report faster than every 30 seconds
- always report at least one per hour (3600 seconds)
- report when temperature changes by 0.2 degrees (which happened in your logs as temp dropped)

**Reply by ChurchRoadEast** [reply]
Ah, very interesting. Is the autobind command generated by Tasmota and if so is it possible to configure the defaults to be something different or is it a case of running this manually on every new pairing?

**Reply by s-hadinger** [reply]
Yes it is it generated by Tasmota and the defaults are in the code and would require to self compile.

**Reply by s-hadinger** [reply]
It's here https://github.com/arendst/Tasmota/blob/d6e71f48e716818218355366f87a441ce38e3a4c/tasmota/xdrv_23_zigbee_8_parsers.ino#L1474

**Reply by ChurchRoadEast** [reply]
Thanks, considering that the zigbee devices would be predominantly used for domestic heating and monitoring and is really optimised for battery life would you say that these defaults are too sensitive? 0.5C and 5% Humidity would probably be granular enough for most heating especially as the four Sonoff devices I have report a range of 0.3C for the same temperature.

**Reply by summongod** [reply]
Hi, thanks for that reply. Shall I close this now?

**Reply by summongod** [reply]
Before I close this, having looked at the code I had a thought that these values ( min_interval,  max_interval, report_change) might be moved to #defines that could be overridden in user_config_override.h? I personally don't like messing inside others code by changing magic numbers in case I cause issues in future builds.

**Reply by s-hadinger** [reply]
I understand, but there are too many values to put them all as #defines. I picked values that seemed reasonable but I'm ready to revisit them.

We can also move to #define some of them.

Are only the threshold you want to change?

**Reply by summongod** [reply]
I see the challenge, perhaps you could have different default settings based upon PowerDescriptor, more conservative for battery powered and less for mains powered?
In my case it was the humidity sensor that was changing quite often so I would change that if anything. Personally I don't think that this sort of sensor requires a huge amount of accuracy but I am sure that others who like graphs that are smoother will disagree.

**Reply by s-hadinger** [reply]
I proposed some constants in `my_user_config.h` in the latest PR, see #10498

**Reply by summongod** [reply]
I just compiled for ESP32 with a CC2530 and I can confirm that the #defines work as expected. Thanks for doing this.

**Reply by s-hadinger** [reply]
Thanks for the confirmation
